core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2003,7 +2000,7 @@ private[spark] class DAGScheduler(
           if (runningStages.contains(stage)) {
             try { // cancelTasks will fail if a SchedulerBackend does not implement killTask
               taskScheduler.cancelTasks(stageId, shouldInterruptTaskThread(job))
-              markStageAsFinished(stage, Some(failureReason))
+              markStageAsFinished(stage, reason)
```

Comment:
If the reason is None, the `completionTime` of the stage would be updated, thus it's not the time that the stage has actually succeed.